aggregator/src/aggregator/aggregation_job_writer.rs

```diff
@@ -593,6 +570,442 @@ impl<const SEED_SIZE: usize, Q: CollectableQueryType, A: vdaf::Aggregator<SEED_S
         )
         .await?;
 
-        Ok(unwritable_report_ids)
+        Ok(indexed.unwritable_report_ids)
+    }
+}
+
+/// Generic callback used in the internals of aggregation job writers. Different implementations are
+/// used when creating new aggregation jobs versus updating existing aggregation jobs.
+trait BatchUpdateCallback {
+    /// This takes one batch and an iterator of aggregation jobs, and returns that batch updated to
+    /// reflect changes due to the aggregation job. Particularly, it will update the
+    /// `outstanding_aggregation_jobs` counter and the `client_timestamp_interval`.
+    fn update_batch<'a, const SEED_SIZE: usize, Q, A, RA>(
+        &self,
+        batch: Batch<SEED_SIZE, Q, A>,
+        aggregation_jobs: impl Iterator<
```

Comment:
nit: I can't find a Rust style guide that discusses it but something bugs me about mixing `impl Trait` notation with a `where` clause in an item. I feel like we should use one or the other.

Reply:
I think I'd prefer to keep the `impl Iterator` for two reasons, firstly because we already have several generic parameters, and I'd like to avoid another. Secondly, the alternative would introduce a type variable on one line, use it on a second line, and bound the type variable on a third, which adds extra indirection when trying to read and understand the method's signature.